hooks/pre/30-upgrade.rb

```diff
@@ -0,0 +1,30 @@
+def migrate_candlepin
+  Kafo::Helpers.execute("/usr/share/candlepin/cpdb --update --password #{Kafo::Helpers.read_cache_data('candlepin_db_password')}")
+end
+
+def migrate_pulp
+  Kafo::Helpers.execute('sudo -u apache pulp-manage-db')
+end
+
+def migrate_foreman
+  Kafo::Helpers.execute('foreman-rake db:migrate')
+end
+
+def upgrade_step(step)
+  noop = app_value(:noop) ? ' (noop)' : ''
+  Kafo::KafoConfigure.logger.info "Upgrade Step: #{step.to_s}" + noop
+  send(step) unless app_value(:noop)
+end
+
+if app_value(:upgrade)
+  say "<%= color('Upgrading...', :info) %>"
```

Comment:
Maybe something more like `Katello upgrading...` or `The application is upgrading...` or `Starting upgrade procedure for application/Katello...`. Perhaps include some output per step? Mine is just showing `Upgrading...` which, as long as it has been proceeding, and without any progress output, I am of the mind it is stuck.